packages/consul/src/main/ts/consul.service.ts

```diff
@@ -0,0 +1,74 @@
+import { Inject, Injectable } from '@nestjs/common'
+import {
+  ConsulDiscoveryService,
+  IConsulKvSetOptions,
+  INormalizedConsulKvValue,
+} from '@qiwi/consul-service-discovery'
+import { IConfig, ILogger, IPromise } from '@qiwi/substrate-types'
+
+export type IConnectionParams = {
+  host: string
+  port: string
+}
+
+const CONSUL_CHECK_REG_INTERVAL = 60000
+
+export interface IConsulService {
+  register(opts: any): IPromise
+  getConnectionParams(opts: any): Promise<IConnectionParams | undefined>
+  getKv(key: string): Promise<INormalizedConsulKvValue>
+}
+
+@Injectable()
+export class ConsulService implements IConsulService {
+  constructor(
+    @Inject('ILogger') private log: ILogger,
+    @Inject('IConfigService') private config: IConfig,
+    @Inject('IDiscoveryService')
+    private discoveryService: ConsulDiscoveryService,
+  ) {}
+
+  async register() {
+    const serviceName: string = this.config.get('name')
+    const consulHost: string = this.config.get('consul.host')
+    const consulPort: string = this.config.get('consul.port')
+    const tags: string[] = this.config.get('consul.tags')
+    const consulToken: string = this.config.get('consul.token')
+    const consulUrl = `http://${consulHost}:${consulPort}/v1/agent/service/register`
+    const port: number = this.config.get('server.port')
+    const ip: string = this.config.get('server.host')
```

Comment:
const consulCheckRegInterval = this.config.get('consul.checkRegInterval') || CONSUL_CHECK_REG_INTERVAL